synapse/storage/databases/main/media_repository.py

```diff
@@ -116,6 +116,56 @@ async def get_local_media(self, media_id: str) -> Optional[Dict[str, Any]]:
             desc="get_local_media",
         )
 
+    async def get_local_media_by_user_paginate(
+        self, start: int, limit: int, user_id: str
+    ) -> Tuple[List[Dict[str, Any]], int]:
+        """Get a paginated list of metadata for a local piece of media
+        which an user_id has uploaded
+
+        Args:
+            start: offset in the list
+            limit: maximum amount of media_ids to retrieve
+            user_id: fully-qualified user id
+        Returns:
+            A paginated list of all metadata of user's media
+        """
+
+        def get_local_media_by_user_paginate_txn(txn):
+            sql_base = """
+                FROM local_media_repository
+                WHERE user_id = ?
+                """
```

Comment:
Honestly I think its clearer to just duplicate these two lines in the two SQL queries